### PR TITLE
KCF Custom Image Tests

### PR DESCRIPTION
**Pull request by sdreyer**
# Description

Added tests for KCF custom images:
- Test client config serialize ~~and de-serialize~~ (We never deserialize on the client side rn)
- Test k8s ~~serialize and~~ (We never serialize on the server side) deserialize
- Test k8sAzureOfflineStoreFactory with a config for ~~blob store~~ (Set to memory store to avoid attempts to connect to azure) with the DockerImage set, get the executor, cast it to KubernetesExecutor, verify that image is set correctly.

### Select type(s) of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

# Checklist:

- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] I have fixed any merge conflicts

### CODE REVIEW

File: client/src/featureform/resources.py [hunk 1]
```diff
@@ -456,29 +456,6 @@ def serialize(self) -> bytes:
         }
         return bytes(json.dumps(config), "utf-8")
 
-
-@typechecked
-@dataclass
-class K8sConfig:
```
Comment (sdreyer) [inline client/src/featureform/resources.py:462]: There were two of these. Removed one

File: provider/k8s_test.go [hunk 2]
```diff
@@ -24,6 +25,82 @@ func uuidWithoutDashes() string {
 	return fmt.Sprintf("a%s", strings.ReplaceAll(uuid.New().String(), "-", ""))
 }
 
+func Test_Deserialize(t *testing.T) {
```
Comment (simba-git) [inline provider/k8s_test.go:28]: No underscore

File: provider/k8s_test.go [hunk 3]
```diff
@@ -24,6 +25,82 @@ func uuidWithoutDashes() string {
...
+	if err != nil {
+		return t.Fatalf("Could not deserialize config: %s", err.Error())
+	}
+
+	if !reflect.DeepEqual(config, expectedConfig) {
+		t.Fatalf("Configs do not match: got %v, expected %v", config, expectedConfig)
+	}
+}
+
+func Test_K8sAzureOfflineStoreFactory(t *testing.T) {
```
Comment (simba-git) [inline provider/k8s_test.go:77]: No Underscore

File: provider/k8s_test.go [hunk 4]
```diff
@@ -24,6 +25,82 @@ func uuidWithoutDashes() string {
...
+	dockerImage := "my-docker-image:latest"
+	KCFConfig := pb.KCFConfig{
```
Comment (simba-git) [inline provider/k8s_test.go:79]: lowercase variable name
More importantly this is unused

**Review by simba-git — APPROVED** [review]

**Pull request merged**